connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -390,6 +390,36 @@ public Schema valueSchema() {
         return valueSchema;
     }
 
+    /**
+     * @param schema the schema to clone the SchemaBuilder from.
+     * @return a new SchemaBuilder from the supplied schema.
+     */
+    public static SchemaBuilder from(Schema schema) {
+        if (null == schema)
+            throw new SchemaBuilderException("schema cannot be null.");
+
+        SchemaBuilder builder = new SchemaBuilder(schema.type());
+        builder.name = schema.name();
+        builder.doc = schema.doc();
+        builder.optional = schema.isOptional();
+        if(null!=schema.parameters()) {
```

Comment:
This is going to complain in checkstyle because of missing spaces around the `if` and `!=`

Reply:
Thanks man! This one has grown in complexity. Now I basically need to be able to clone a struct as well.